src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -329,41 +350,82 @@ static async Task<HttpResponseMessage> Impl(HttpRequestMessage request, Cancella
         }
     }
 
-    internal sealed class ReadableStreamPullState
+    internal sealed class WasmHttpWriteStream : Stream
     {
-        private readonly Stream _stream;
-        private readonly CancellationToken _cancellationToken;
-        private readonly byte[] _buffer;
+        private readonly JSObject _transformStream;
 
-        public ReadableStreamPullState(Stream stream, CancellationToken cancellationToken)
+        public WasmHttpWriteStream(JSObject transformStream)
         {
-            ArgumentNullException.ThrowIfNull(stream);
+            ArgumentNullException.ThrowIfNull(transformStream);
 
-            _stream = stream;
-            _cancellationToken = cancellationToken;
-            _buffer = new byte[65536];
+            _transformStream = transformStream;
         }
 
-        public async Task PullAsync()
+        private async Task WriteAsyncCore(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken)
         {
-            try
-            {
-                int length = await _stream.ReadAsync(_buffer, _cancellationToken).ConfigureAwait(true);
-                ReadableStreamControllerEnqueueUnsafe(this, _buffer, length);
-            }
-            catch (Exception ex)
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
since we switched from pull to push model, this call could come from any thread. And so it will need to be dispatch to the thread/worker which has the JS objects on it.

Reply:
At this point we are only doing two `abortController.Dispose` and `transformStream.Dispose` calls which already dispatch to their `SynchronisationContext`. Also we do use `ConfigureAwait(true)`. Or do will still need to dispatch? Is it only required for `FEATURE_WASM_THREADS`?

E.g.
```c#
#if FEATURE_WASM_THREADS
 transformStream.SynchronizationContext.Send(...);
```

Reply:
I mean `TransformStreamWriteUnsafe` is pushing bytes, so wrapping it with `FEATURE_WASM_THREADS` and dispatch should be necessary I think. 

The `FEATURE_WASM_THREADS` is the multi-threaded build of runtime. 
You can build it with `/p:MonoWasmBuildVariant=multithread` 
Make sure to `rm -rf artifacts` before doing so.

Reply:
Or you can keep streaming not implemented in multi-threaded build and return false for `SupportsStreamingRequest` under `FEATURE_WASM_THREADS`

Reply:
and throw here, just to be sure.

Reply:
> I mean TransformStreamWriteUnsafe is pushing bytes, so wrapping it with FEATURE_WASM_THREADS and dispatch should be necessary I think.

Sorry, I was looking at the wrong code.
Yes, it is indeed needed and with https://github.com/dotnet/runtime/pull/91699/commits/0f2ee8f47beb7d92a015c9e3b15489c6d45670d5 the tests passed when building / running with `/p:MonoWasmBuildVariant=multithread` otherwise they would fail.